File: docs/examples/python/run_samples.py
import os
import sys

from fetch_jwks import fetch_jwks
from verify_token import verify_token


def main() -> None:
    try:
        jwks = fetch_jwks()
        print("Fetched JWKS:", jwks)
    except Exception as exc:
        print("Failed to fetch JWKS:", exc)
        sys.exit(1)

    token = None
    if len(sys.argv) > 1:
        token = sys.argv[1]
    else:
        token = os.environ.get("CORE_AUTH_SAMPLE_TOKEN")

    if not token:
        print("No token provided. Pass one as an argument or set CORE_AUTH_SAMPLE_TOKEN.")
        return

    try:
        verification = verify_token(token)
        print("Verification response:", verification)
    except Exception as exc:
        print("Failed to verify token:", exc)
        sys.exit(1)


if __name__ == "__main__":
    main()

File: docs/examples/node/run_samples.ts
import { fetchJwks } from './fetch-jwks';
import { verifyToken } from './verify-token';

async function main() {
  try {
    const jwks = await fetchJwks();
    console.log('Fetched JWKS:', JSON.stringify(jwks, null, 2));
  } catch (err) {
    console.error('Failed to fetch JWKS:', err);
    process.exitCode = 1;
    return;
  }

  const tokenFromCli = process.argv[2];
  const tokenFromEnv = process.env.CORE_AUTH_SAMPLE_TOKEN;
  const token = tokenFromCli ?? tokenFromEnv;

  if (!token) {
    console.log('No token provided. Set CORE_AUTH_SAMPLE_TOKEN or pass one as an argument to verify.');
    return;
  }

  try {
    const verification = await verifyToken(token);
    console.log('Verification response:', JSON.stringify(verification, null, 2));
  } catch (err) {
    console.error('Failed to verify token:', err);
    process.exitCode = 1;
  }
}

main();
